drivers/modem/ublox-sara-n310.c

```
@@ -0,0 +1,526 @@
+
+		/* get and store modem info */
+		SETUP_CMD("AT+CGMI", "", on_cmd_atcmdinfo_manufacturer, 0U, ""),
+		SETUP_CMD("AT+CGMM", "", on_cmd_atcmdinfo_model, 0U, ""),
+		SETUP_CMD("AT+CGMR", "", on_cmd_atcmdinfo_revision, 0U, ""),
+		SETUP_CMD("AT+CGSN", "", on_cmd_atcmdinfo_imei, 0U, ""),
+
+		/* enable functionality */
+		SETUP_CMD_NOHANDLE("AT+CFUN=1"),
+	};
+
+	/* reset module */
+	ret = modem_cmd_send(&mdata.context.iface, &mdata.context.cmd_handler,
+			     NULL, 0, "AT+CFUN=16", &mdata.sem_response,
+			     K_SECONDS(5));
+
+	if (ret < 0) {
+		LOG_ERR("module reset failed: %d", ret);
+		return;
+	}
+
+	modem_cmd_send(&mdata.context.iface, &mdata.context.cmd_handler, NULL,
+		       0, "AT+CCID", &mdata.sem_response, K_SECONDS(1));
+
+	/* give modem time to start responding after restart */
+	ret = -1;
+
+	while (counter++ < 50 && ret < 0) {
+		k_sleep(K_SECONDS(2));
+		ret = modem_cmd_send(&mdata.context.iface,
+				     &mdata.context.cmd_handler, NULL, 0, "AT",
+				     &mdata.sem_response, K_SECONDS(5));
+		if (ret < 0 && ret != -ETIMEDOUT) {
+			break;
+		}
+	}
+
+	if (ret < 0) {
+		LOG_ERR("Modem wait loop error: %d", ret);
+		return;
+	}
+
+	/* send setup commands */
+	ret = modem_cmd_handler_setup_cmds(&mdata.context.iface,
+					   &mdata.context.cmd_handler,
+					   setup_cmds, ARRAY_SIZE(setup_cmds),
+					   &mdata.sem_response, K_SECONDS(5));
+
+	if (ret < 0) {
+		LOG_ERR("Modem setup cmds error: %d", ret);
+		return;
+	}
+
+	/* register operator automatically */
+	ret = modem_cmd_send(&mdata.context.iface, &mdata.context.cmd_handler,
+			     NULL, 0, "AT+COPS=0", &mdata.sem_response,
+			     K_SECONDS(120));
+
+	if (ret < 0) {
+		LOG_ERR("AT+COPS error: %d", ret);
+		return;
+	}
+
+	close_socket();
+
+	/* re-init underlying UART comms */
+	ret = modem_iface_uart_init_dev(&mdata.context.iface,
+					CONFIG_MODEM_UBLOX_SARA_N310_UART_NAME);
+
+	if (ret < 0) {
+		LOG_ERR("uart re-init failed");
+		return;
+	}
+
+	LOG_INF("Modem is ready.");
+}
```

is it possible to make this function return an int so you can return an error code?